ext/symengine/ruby_ntheory.c

```diff
@@ -0,0 +1,105 @@
+#include "ruby_ntheory.h"
+
+VALUE cntheory_onearg(void (*cwfunc_ptr)(basic_struct*, const basic_struct*), VALUE operand1) {
```

Comment:
@isuruf yes.. it is. 
Should I use that instead? 
Or should I make another source file with methods common to different parts?

What's the better way?